tests/apollo/test_skvbc_preexecution.py

```diff
@@ -94,49 +95,48 @@ async def run_concurrent_pre_execution_requests(self, skvbc, clients, num_of_req
     @unittest.skip("unstable")
     @with_trio
     @with_bft_network(start_replica_cmd)
-    async def test_sequential_pre_process_requests(self, bft_network):
+    @verify_linearizability(pre_exec_enabled=True)
+    async def test_sequential_pre_process_requests(self, bft_network, tracker):
         """
         Use a random client to launch one pre-process request in time and ensure that created blocks are as expected.
         """
         bft_network.start_all_replicas()
-        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
 
         for i in range(NUM_OF_SEQ_WRITES):
             client = bft_network.random_client()
-            await self.send_single_write_with_pre_execution(skvbc, client)
+            await tracker.send_tracked_write(client, 2)
 
     @unittest.skip("unstable")
     @with_trio
     @with_bft_network(start_replica_cmd)
-    async def test_concurrent_pre_process_requests(self, bft_network):
+    @verify_linearizability(pre_exec_enabled=True)
+    async def test_concurrent_pre_process_requests(self, bft_network, tracker):
         """
         Launch concurrent requests from different clients in parallel. Ensure that created blocks are as expected.
         """
         bft_network.start_all_replicas()
-        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
 
         clients = bft_network.random_clients(MAX_CONCURRENCY)
         num_of_requests = len(clients)
-        sent_count = await self.run_concurrent_pre_execution_requests(skvbc, clients, num_of_requests)
-        self.assertTrue(sent_count >= num_of_requests)
+        rw = await tracker.run_concurrent_ops(num_of_requests, write_weight=0.9)
+        self.assertTrue(rw[0] + rw[1] >= num_of_requests)
 
     @with_trio
     @with_bft_network(start_replica_cmd)
-    async def test_long_time_executed_pre_process_request(self, bft_network):
+    @verify_linearizability(pre_exec_enabled=True)
+    async def test_long_time_executed_pre_process_request(self, bft_network, tracker):
         """
         Launch pre-process request with a long-time execution and ensure that created blocks are as expected
         and no view-change was triggered.
         """
         bft_network.start_all_replicas()
-        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
 
         client = bft_network.random_client()
         client.config = client.config._replace(req_timeout_milli=LONG_REQ_TIMEOUT_MILLI, retry_timeout_milli=1000)
-        await self.send_single_write_with_pre_execution(skvbc, client, long_exec=True)
+        await tracker.send_tracked_write(client, 2, long_exec=True)
 
-        clients = bft_network.clients.values()
         with trio.move_on_after(seconds=1):
-            await self.send_indefinite_pre_execution_requests(skvbc, clients)
```

Comment:
Can we delete `send_indefinite_pre_execution_requests` ?

Reply:
We can, but don't you think we should not remove this code to a case we will run a test without the tracker?

Reply:
agreed, thanks.